### ISSUE TITLE
[TEST] Rest API tests pass when they should be failing

### ISSUE THREAD

**Issue by HonzaKral** [elastic/elasticsearch#27126]
in https://github.com/elastic/elasticsearch/blob/5.6/rest-api-spec/src/main/resources/rest-api-spec/test/indices.shrink/20_source_mapping.yml#L37-L47 you can see an equivalent of 
```
curl -X PUT localhost:9200/_template/tpl1 -d '{
    "index_patterns": "targ*",
    "mappings": {
        "test": {
            "properties": {
                "count": {
                    "type": "integer"
                }
            }
        }
    }
}'
```

which is not a valid command yet this test has been part of several releases. Either the test failure is being ignored or the test runner is wrong.

**Comment by rjernst** [comment]
Can you elaborate on what is wrong with this request?

**Comment by HonzaKral** [comment]
it doesn't have the `template` key in the body as is required for index templates. Executing it on `5.6` cluster produces an error:
```
{"error":{"root_cause":[{"type":"action_request_validation_exception","reason":"Validation Failed: 1: template is missing;"}],"type":"action_request_validation_exception","reason":"Validation Failed: 1: template is missing;"},"status":400}```

**Comment by dakrone** [comment]
I think we should have something (a gradle task maybe?) that fails hard if there are `.yml` files in t he REST stuff, telling people to convert them to `.yaml` files

**Comment by HonzaKral** [comment]
sonds like it was addressed in https://github.com/elastic/elasticsearch/pull/24740

Maybe we can standardize on this since all the clients run bot `yml` and `yaml` since both are valid yaml extensions and do both?

**Comment by javanna** [comment]
I am going to push a fix to the 5.6 branch. Note that in 5.6 we only run `.yaml` files, while in later branches we only run `.yml`. Discussion on this happened as part of #24633 . See also #24659. Given that we now enforce that `.yaml` files cannot be there (change not backported to 5.6) we are good.

**Comment by javanna** [comment]
Fixed with 9188c9cac4bc77628970a7d4c8dc8a3868e993f9
